### PR TITLE
[AV-1834, AV-1810] Add key list

### PR DESCRIPTION
**Pull request by dasconnor**
Adds `key list` command. I also made an improvement so that the `subnet list` command won't get tripped up by temporary files.

### CODE REVIEW

File: tests/e2e/key/suite.go [hunk 1]
```diff
@@ -117,4 +117,36 @@ var _ = ginkgo.Describe("[Key]", func() {
...
+		}
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// The matcher for this test is a little weird. Instead of matching an exact
+		// string, we check that it contains all of a few necessary substrings. This
+		// is to facilitate running the test locally. If you have other keys in your
+		// key directory, they will be printed as well. It's impossible to check the
+		// list output for exact equality without removing pre-existing user keys.
+		// Hence, the matcher here.
+		gomega.Expect(output).Should(gomega.ContainSubstring(expectedLine1))
```
Comment (felipemadero) [inline tests/e2e/key/suite.go:148]: probably you may use a regex to include all 3 things in one check. also a regex will avoid problems related to key length (if there is any)

File: tests/e2e/key/suite.go [hunk 2]
```diff
@@ -117,4 +117,36 @@ var _ = ginkgo.Describe("[Key]", func() {
 		_, err = commands.CreateKey(keyName)
 		gomega.Expect(err).Should(gomega.HaveOccurred())
 	})
+
+	ginkgo.It("can list a created key", func() {
+		expectedLine1 := "+----------+"
+		expectedLine2 := "| KEY NAME |"
+		expectedLine3 := "| e2eKey   |"
```
Comment (felipemadero) [inline tests/e2e/key/suite.go:124]: are you going to have problems with different key lengths?

File: tests/e2e/key/suite.go (hunk 1 above)
Comment (dasconnor) [reply tests/e2e/key/suite.go]: updated, I check a regex and that the expected key is in there

File: cmd/keycmd/list.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	if err != nil {
+		return err
+	}
+
+	rows := [][]string{}
+
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), constants.KeySuffix) {
+			filename := f.Name()
+			rows = append(rows, []string{filename[:len(filename)-len(constants.KeySuffix)]})
```
Comment (holisticode) [inline cmd/keycmd/list.go:42]: Note: there is the `strings.TrimSuffix` method which makes it a bit more readable, but, on the other hand, slower, so up to you.

File: cmd/keycmd/list.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	files, err := os.ReadDir(app.GetKeyDir())
+	if err != nil {
+		return err
+	}
+
+	rows := [][]string{}
+
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), constants.KeySuffix) {
+			filename := f.Name()
```
Comment (holisticode) [inline cmd/keycmd/list.go:41]: We could also do some validation here to make sure it's actually a private key and not something (veeeery accidentally....) with the suffix. Leaving to your judgement too.

File: tests/e2e/key/suite.go [hunk 5]
```diff
@@ -117,4 +117,33 @@ var _ = ginkgo.Describe("[Key]", func() {
...
+			utils.PrintStdErr(err)
+		}
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// The matcher for this test is a little weird. Instead of matching an exact
+		// string, we check that it matches a regex and contains created keyName. This
+		// is to facilitate running the test locally. If you have other keys in your
+		// key directory, they will be printed as well. It's impossible to check the
+		// list output for exact equality without removing pre-existing user keys.
+		// Hence, the matcher here.
```
Comment (holisticode) [inline tests/e2e/key/suite.go:145]: Note: if you were able to have the keys created in a tmp test dir, like we do with unit tests, it _would_ be possible. However, this being e2e tests, I understand that this probably amounts to unnecessary overhead. But worth having if possible.

**Review by holisticode — COMMENTED** [review]
Basically all good for me. I left some notes which do not require intervention, unless you decide to.

File: cmd/keycmd/list.go (hunk 4 above)
Comment (dasconnor) [reply cmd/keycmd/list.go:41]: Eh, I don't think this is really worth it. I don't care so much about users trying to fuzz test us with bad inputs. There are a few other places where we make similar assumptions.

File: cmd/keycmd/list.go (hunk 3 above)
Comment (dasconnor) [reply cmd/keycmd/list.go]: updated

**Review by holisticode — APPROVED** [review]

**Pull request merged**